apps/docs/pages/docs/integrating-puck/rich-text-editing.mdx

```diff
@@ -0,0 +1,342 @@
+import { ConfigPreview, PuckPreview } from "@/docs/components/Preview";
+import { Puck, RichTextMenu } from "@/puck";
+import { Callout } from "nextra/components";
+
+# Rich Text Editing
+
+Puck supports rich text editing via the [`richtext`](/docs/api-reference/fields/richtext) field.
+
+## Adding a rich text field
+
+To use rich text, add the `richtext` field to your component config:
+
+```tsx copy showLineNumbers {6}
+const config = {
+  components: {
+    Example: {
+      fields: {
+        body: {
+          type: "richtext",
+        },
+      },
+      render: ({ body }) => body,
+    },
+  },
+};
+```
+
+<ConfigPreview
+  label="Example"
+  componentConfig={{
+    fields: {
+      body: {
+        type: "richtext",
+      },
+    },
+    defaultProps: {
+      body: "<p>Hello, <strong>world</strong></p><ul><li><p>List item 1</p></li><li><p>List item 2</p></li><li><p>List item 3</p></li></ul>",
+    },
+    render: ({ body }) => {
+      return body;
+    },
+  }}
+>
+  <Puck.Preview />
+</ConfigPreview>
+
+## Enabling inline editing
+
+Set [`contentEditable`](/docs/api-reference/fields/richtext#contenteditable) to make the field editable inline:
+
+```tsx copy showLineNumbers {7}
+const config = {
+  components: {
+    Example: {
+      fields: {
+        body: {
+          type: "richtext",
+          contenEditable: true,
+        },
+      },
+      render: ({ body }) => body,
+    },
+  },
+};
+```
+
+<ConfigPreview
+  label="Example"
+  componentConfig={{
+    fields: {
+      body: {
+        type: "richtext",
+        contentEditable: true,
+      },
+    },
+    defaultProps: {
+      body: "<p>Edit me <strong>inline:</strong></p><ul><li><p>List item 1</p></li><li><p>List item 2</p></li><li><p>List item 3</p></li></ul>",
+    },
+    render: ({ body }) => {
+      return body;
+    },
+  }}
+  hideActionBar={false}
+>
+  <Puck.Preview />
+</ConfigPreview>
+
+## Customizing behavior
+
+The rich text field is built with [Tiptap](https://tiptap.dev), and comes with [several Tiptap extensions](/docs/api-reference/fields/richtext#included-extensions) pre-configured. Editor behavior can be customized by configuring these extensions.
+
+### Disabling functionality
+
+Disable any extension by setting it to `false` in [`options`](/docs/api-reference/fields/richtext#options):
+
+```ts showLineNumbers copy {7-10}
+const config = {
+  components: {
+    Example: {
+      fields: {
+        body: {
+          type: "richtext",
+          options: {
+            // Disable bold extension
+            bold: false,
+          },
+        },
+      },
+      render: ({ body }) => body,
+    },
+  },
+};
+```
+
+<ConfigPreview
+  label="Example"
+  componentConfig={{
+    fields: {
+      body: {
+        type: "richtext",
+        options: { bold: false },
+      },
+    },
+    defaultProps: {
+      body: "<p>Hello, <strong>world</strong></p>",
```

Comment:
I am confused by this not sure if it's a gap in my understanding or the docs.  If bold is disabled shouldn't all of bold be disabled not just the control?

Reply:
It's a bug. We weren't including the options or extensions in the sanitisation for the rendered output. Good catch!